### PR TITLE
Updates for April and May 2023 releases

### LINKED ISSUE #27: [RELEASE] KES 2023-04-03T16-41-28Z
KES [2023-04-03T16-41-28Z](https://github.com/minio/kes/releases/tag/2023-04-03T16-41-28Z) has changes that impact the docs:

- [x] An API section has been added to the server.yaml config file ([Commit dd083456](https://github.com/minio/kes/commit/dd083456b99964d43db04ce38cd678693982f3c8))
- [x] Keystore status is now exposed in the `/v1/status` API ([Commit ffbbe404](https://github.com/minio/kes/commit/ffbbe404d9f5053ccce86a15fd8ffcc68d5d677e))

### LINKED ISSUE #30: [RELEASE]  2023-05-02T22-48-10Z
Release [2023-05-02T22-48-10Z](https://github.com/minio/kes/releases/tag/2023-05-02T22-48-10Z) has changes that impact the docs:

- [x] Allow readiness probe to skip auth in K8s environments ([Commit `d15990d`](https://github.com/minio/kes/commit/d15990de49738ab12e6af66fe29ddc9b8e88258a))
- [x] Improve `kes identity new` command ([Commit `8a9e169`](https://github.com/minio/kes/commit/8a9e169ba56154f7e71e058a4deb2ce6ac4f00a3))

### PR DESCRIPTION
**Pull request by djwfyi**
Updates for two KES releases
    
`KES 2023-04-03T16-41-28Z`
- Adds new API section to configuration doc
- Creates a new include for the server.yaml example
- Updates the status API with additional info
    
Closes #27
    
`KES 2023-05-02T22-48-10Z`
- Updates kes identity new command
- Adds ready API endpoint
    
Closes #30

Staged:

*Note: If you click around the site with the generated links, you'll need to manually add `index.html` to the end of each URL.*

- [API Config](http://192.241.195.202:9000/staging/api-updates/tutorials/configuration/index.html#api-configuration)
- [Server](http://192.241.195.202:9000/staging/api-updates/includes/server-config/index.html)
- [Ready API](http://192.241.195.202:9000/staging/api-updates/concepts/server-api/index.html#ready)
- [Status API](http://192.241.195.202:9000/staging/api-updates/concepts/server-api/index.html#status)

### CODE REVIEW

File: content/concepts/server-api.md [hunk 1]
```diff
@@ -8,9 +8,18 @@ tableOfContents: true
 
 ## API Overview
 
+By default, KES requires a valid certificate for any API call.
+Requests without a certificate fail with a TLS error.
+
+You can disable the requirement for a valid certificate when calling some endpoints in the KES [configuration file]({{< relref "tutorials/configuration.md#api-configuration" >}}).
+
+If any endpoint does not require a certificate, failed calls result in an HTTP error instead of a TLS error.
```
Comment (feorlen) [inline content/concepts/server-api.md:16]: This seems a bit ambiguous to me, would something like this be accurate:
> Endpoints that do not require a certificate return an HTTP error for a failed call instead of a TLS error.

~I suspect other (certificate required) endpoints still return TLS errors on failure.~
Seems not requiring a cert for one _does_ affect others. I figured it out below, but was confused  when I saw this first.

File: content/concepts/server-api.md [hunk 2]
```diff
@@ -72,6 +81,42 @@ $ curl \
...
+$ curl \
+    --key root.key \
+    --cert root.cert \
+    --request GET \
+    'https://play.min.io:7373/v1/ready'
+```
+
+#### Sample Response
+
+Request results in either a ``200 Ok`` response or an error.
```
Comment (feorlen) [inline content/concepts/server-api.md]: Might this be `A request`? 

But perhaps an alternative:
> The request returns either `200 OK` or an error.
(This presumes the reader is familiar enough with REST APIs that they know `200` is a response code.)

Do we know what error it returns?

File: content/includes/server-config.md [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+    # about the client to the KES server.
+    header:
+      # The HTTP header containing the URL-escaped and PEM-encoded
+      # certificate of the kes client forwarded by the TLS proxy.
+      cert: X-Tls-Client-Cert
+
+# The API configuration. The APIs exposed by the KES server can
+# be adjusted here. Each API is identified by its API path.
+#
+# In general, the KES server uses sane defaults for all APIs.
```
Comment (feorlen) [inline content/includes/server-config.md]: Maybe swap `sane` for `reasonable`?

File: content/includes/server-config.md [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+# Only customize the APIs if there is a real need.
+# 
+# Disabling authentication for an API must be carefully evaluated.
+# One example, when disabling authentication may be justified, would
+# be the liveness and readiness probes in a Kubernetes environment.
```
Comment (feorlen) [inline content/includes/server-config.md]: This sentence is a little convoluted, perhaps:
```
# For example, you may choose to disable authentication for a
# Kubernetes readiness probe, but not metrics.

(I'm guessing "Kubernetes readiness probe" is `v1/ready`?)

Comment (feorlen) [reply content/includes/server-config.md]: Oh this is directly from the original in the repo. Hmm.

File: content/tutorials/configuration.md [hunk 5]
```diff
@@ -113,6 +114,42 @@ tls:
...
 
+#### API Configuration
+
+By default, KES requires a valid TLS certificate for all API calls.
+You can change this behavior for specific API endpoints to allow requests to the endpoints without a certificate.
+
+{{< admonition type="caution" >}}
+The default behavior of the KES API endpoints should suffice in most every situation.
+Only customize endpoints for specific need
+{{< /admonition >}}
```
Comment (feorlen) [inline content/tutorials/configuration.md:125]: Slightly shorter alternative?
> The default behavior of the KES API endpoints should be suitable for most situations. Only customize endpoints for a specific need.

File: content/tutorials/configuration.md [hunk 6]
```diff
@@ -113,6 +114,42 @@ tls:
...
+
+When you disable authentication for at least one endpoint, KES no longer requires a certificate for a client to call _any_ API endpoint.
```
Comment (feorlen) [inline content/tutorials/configuration.md:127]: Ohhhh this is why I was confused about when the cert is required. Making cert not required for one _does_ change behavior for others. Hence my other comment. Maybe this could be more prominent. (Admonition seems a little much however.)

File: content/tutorials/configuration.md [hunk 7]
```diff
@@ -113,6 +114,42 @@ tls:
...
+However, the request must still include a certificate for KES to successfully process a call to an API endpoint that does require authentication.
+
+This change means that on KES servers with at least one endpoint configured to disable authentication, clients no receive a TLS error on failure, but an HTTP error instead.
```
Comment (feorlen) [inline content/tutorials/configuration.md]: `clients do not receive a TLS error` ?

File: content/tutorials/configuration.md [hunk 8]
```diff
@@ -287,8 +324,11 @@ policy:
     - ${MY_APP_IDENTITY}
 ```
 
-**Note:** The server does not fail if an environment variable is not present or not a "valid" identity. 
+{{< admonition title="Admonition title" type="note" >}}
+The server does not fail if an environment variable is not present or not a "valid" identity.
```
Comment (feorlen) [inline content/tutorials/configuration.md]: I'm not exactly sure what `does not fail` entails, does this mean the server silently continues on as if nothing happened? Is something logged? Also the multiple negations in this sentence make it a bit hard to follow.

File: content/tutorials/configuration.md [hunk 9]
```diff
@@ -320,4 +360,8 @@ keystore:
     path: ./keys # The key store directory. Keys will be created inside ./keys/
 ```
 
-For production setups, only secure key stores backed by a KMS such as [Hashicorp Vault](https://github.com/minio/kes/wiki/Hashicorp-Vault-Keystore). 
\ No newline at end of file
+For production setups, only secure key stores backed by a KMS such as [Hashicorp Vault](https://github.com/minio/kes/wiki/Hashicorp-Vault-Keystore).
```
Comment (feorlen) [inline content/tutorials/configuration.md]: Maybe this was intended to be `only use secure key stores` ?

**Review by feorlen — CHANGES_REQUESTED** [review]
Noted a couple typos and offered some edit ideas for your consideration. 

The bit where making cert not required for one endpoint results in behavior change to other endpoints was fuzzy for me on first read. Does the impact of that setting need to be more prominent? Granted I don't know much about this topic, but it was unexpected.

File: content/concepts/server-api.md (hunk 2 above)
Comment (djwfyi) [reply content/concepts/server-api.md]: Replacing with a table of possible HTML responses.

File: content/includes/server-config.md (hunk 3 above)
Comment (djwfyi) [reply content/includes/server-config.md]: Ooh, really glad you caught that. Definite replacement here. 😱

File: content/includes/server-config.md (hunk 4 above)
Comment (djwfyi) [reply content/includes/server-config.md]: Cleaned it up in ours anyway.

File: content/tutorials/configuration.md (hunk 5 above)
Comment (djwfyi) [reply content/tutorials/configuration.md:125]: Not much shorter, but probably clearer. Going with your suggestion.

File: content/tutorials/configuration.md (hunk 6 above)
Comment (djwfyi) [reply content/tutorials/configuration.md:127]: I agree that an admonition is not needed here. Going to leave as is for now unless an idea for highlighting it further comes to mind.

File: content/tutorials/configuration.md (hunk 7 above)
Comment (djwfyi) [reply content/tutorials/configuration.md]: Yes, that.

File: content/tutorials/configuration.md (hunk 8 above)
Comment (djwfyi) [reply content/tutorials/configuration.md]: Reversed the lines in the admonition and reworded line 328 to clear it up a bit.

File: content/tutorials/configuration.md (hunk 9 above)
Comment (djwfyi) [reply content/tutorials/configuration.md]: Cleaned up

File: content/tutorials/configuration.md [hunk 10]
```diff
@@ -120,14 +120,14 @@ By default, KES requires a valid TLS certificate for all API calls.
 You can change this behavior for specific API endpoints to allow requests to the endpoints without a certificate.
 
 {{< admonition type="caution" >}}
-The default behavior of the KES API endpoints should suffice in most every situation.
-Only customize endpoints for specific need
+The default behavior of the KES API endpoints should be suitable for most situations.
+Only customize endpoints for specific need.
```
Comment (feorlen) [inline content/tutorials/configuration.md]: teeny typo: `for a specific need.`

**Review by feorlen — APPROVED** [review]
Flagged a teeny tiny typo, otherwise LGTM. Table of HTTP responses is 💯
